numpy/ma/core.py

```diff
@@ -5213,7 +5213,8 @@ def round(self, decimals=0, out=None):
             out.__setmask__(self._mask)
         return out
 
-    def argsort(self, axis=None, kind='quicksort', order=None, fill_value=None):
+    def argsort(self, axis=-1, kind='quicksort', order=None,
+                endwith=True, fill_value=None):
```

Comment:
Regarding named argument order, perhaps we should enforce that `endwith` is a boolean, just to help guard against someone (stupidly) passing unnamed arguments, and assuming they match the order of `sort`?